File: ASP Core - Backend/Models/Author.cs
namespace aspcorebackend.Models {
    public class Author {
        public int AuthorId { get; set; }
        public string Firstname { get; set; } = null!;
        public string Lastname { get; set; } = null!;
        public DateTime AddedAt { get; set; }
        public List<Book> Books { get; set; } = [];

        public Author() {
            this.AddedAt = DateTime.Now;
        }
    }
}

File: Spring Boot - Backend/src/main/java/com/booknav/springbackend/entities/Author.java
package com.booknav.springbackend.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "authors")
@Getter
@Setter
public class Author {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "author_id")
    private int authorId;
    @Column(name = "firstname")
    private String firstName;
    @Column(name = "lastname")
    private String lastName;
    @OneToMany(mappedBy = "author")
    private List<Book> books;
    private LocalDateTime addedAt;

    public Author() {
        this.addedAt = LocalDateTime.now();
    }

    public Author(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.addedAt = LocalDateTime.now();
    }

    public void addBook(Book theBook) {
        if (books == null) {
            books = new ArrayList<>();
        }
        books.add(theBook);
        theBook.setAuthor(this);
    }
}
